client/network/sync/src/block_request_handler.rs

```diff
@@ -261,9 +258,22 @@ where
 				}
 			}
 
+			Some(block_response)
+		} else {
+			None
+		};
+
+		debug!(
+			target: LOG_TARGET,
+			"Sending result of block request from {peer} starting at `{from_block_id:?}`: \
```

Comment:
This should be put into the `if` branch? Because we don't send any block response in the `None` case? We can instead add a second logging that says that we don't send any response.

Reply:
Hmm, actually we still send a response in terms of the `OutgoingResponse` below, which contains a result `Err(())` in the case of `None`. `blocks` will be displayed as `Some(100)` or `None` which indicates we send 100 blocks or we don't send back any blocks, should be clear enough for the diagnostic purpose IMO.